packages/beacon-node/src/api/rest/base.ts

```diff
@@ -97,7 +97,12 @@ export class RestApiServer {
       if (err instanceof ErrorAborted || err instanceof NodeIsSyncing) return;
 
       const {operationId} = res.context.config as RouteConfig;
-      this.logger.error(`Req ${req.id} ${operationId} error`, {}, err);
+
+      if (err instanceof ApiError && err.statusCode < 500) {
```

Comment:
Are you sure that all **400 codes** should not be logged as errors? I would think particular instances such as a [404 for resolving a block ID](https://github.com/ChainSafe/lodestar/blob/unstable/packages/beacon-node/src/api/impl/beacon/blocks/utils.ts#L34) it should be thrown as an error, otherwise we can't fetch a block root for example.

Also [if we fail to provide a set of sync committee duties for a particular epoch](https://github.com/ChainSafe/lodestar/blob/unstable/packages/beacon-node/src/api/impl/validator/index.ts#L479) , a `400` error is thrown. Perhaps what could be considered is optionally returning the stack trace as suggested in the [spec](https://github.com/ethereum/beacon-APIs/blob/master/types/http.yaml#L141)

Reply:
> Are you sure that all 400 codes should not be logged as errors?

pretty much, in fact, could even remove the `err.statusCode < 500` in my opinion, there is anyway just one error at the moment were this would make a difference https://github.com/ChainSafe/lodestar/blob/cd2ec2ee9878a12c09742c76fb2681a4e7d551e2/packages/beacon-node/src/api/impl/validator/index.ts#L356

> [404 for resolving a block ID](https://github.com/ChainSafe/lodestar/blob/unstable/packages/beacon-node/src/api/impl/beacon/blocks/utils.ts#L34) it should be thrown as an error, otherwise we can't fetch a block root for example.

I am not really sure how this changes affects what you mentioned, all that changed now is that we log it as a warning with less verbosity (no stacktrace) instead of an error. The API response is still the same.
The two examples you linked just further enforce the change I want to introduce, there is no reason at all we should log a stacktrace when those errors happen.


> Perhaps what could be considered is optionally returning the stack trace as suggested in the [spec](https://github.com/ethereum/beacon-APIs/blob/master/types/http.yaml#L141)

For some unexpected errors it could make sense to also include the stacktrace in the HTTP error response but might as well just look at the logs instead. Maybe someone else in the team as some opinions on this, but I would not implement this unless there is a specific use case for it.